src/main/java/com/google/ehub/utility/UserRecommendationUtils.java

```diff
@@ -15,18 +16,55 @@
 
   /**
    * Finds the most recommended emails up to a maximum of ten recommendations in descending order.
+   * In the case of a tie, the order will use increasing lexicographical order.
    *
    * @param itemLikes map with the key representing an itemId and the value representing the emails
    *     that liked that item
    * @return list containing the most recommended emails in descending order
    */
   public List<String> getRecommendedEmails(Map<Long, Set<String>> itemLikes) {
+    return getRecommendedEmailsInDescendingOrder(getEmailPriorityQueue(getEmailFreqs(itemLikes)));
+  }
+
+  private Map<String, Integer> getEmailFreqs(Map<Long, Set<String>> itemLikes) {
+    Map<String, Integer> emailFreqs = new HashMap<String, Integer>();
+
+    for (Map.Entry<Long, Set<String>> emailGroup : itemLikes.entrySet()) {
+      for (String email : emailGroup.getValue()) {
+        emailFreqs.merge(email, 1, (oldFreq, deltaFreq) -> oldFreq + deltaFreq);
+      }
+    }
+
+    return emailFreqs;
+  }
+
+  private MinMaxPriorityQueue<Map.Entry<String, Integer>> getEmailPriorityQueue(
+      Map<String, Integer> emailFreqs) {
+    MinMaxPriorityQueue<Map.Entry<String, Integer>> emailPriorityQueue =
+        MinMaxPriorityQueue
+            .orderedBy((Map.Entry<String, Integer> a, Map.Entry<String, Integer> b) -> {
+              // If the counts are tied, then compare emails lexicographically.
+              if (a.getValue() == b.getValue()) {
+                return a.getKey().compareTo(b.getKey());
+              }
+
+              return Integer.compare(b.getValue(), a.getValue());
+            })
+            .maximumSize(MAX_NUMBER_OF_RECOMMENDATIONS)
+            .create();
+
+    emailPriorityQueue.addAll(emailFreqs.entrySet());
+
+    return emailPriorityQueue;
+  }
+
+  private List<String> getRecommendedEmailsInDescendingOrder(
+      MinMaxPriorityQueue<Map.Entry<String, Integer>> emailPriorityQueue) {
     List<String> recommendedEmails = new ArrayList<String>();
 
-    // TODO: Implement algorithm that finds the users that have the highest amount of common items,
-    // could be implemented
-    // by using a Map<String, int> storing counts and a priority queue to get the
-    // "maxNumberOfRecommendations"
+    while (!emailPriorityQueue.isEmpty()) {
+      recommendedEmails.add(emailPriorityQueue.poll().getKey());
+    }
 
     return recommendedEmails;
   }
```

Comment:
I think this entire function can be replaced with
` FluentIterable.from(emailPriorityQueue).transform(input -> input.getKey()).toList();`

Then you can move that one-liner into `getEmailPriorityQueue()` and have it return a List.
Or inside `getRecommendedEmails()` you can do something like below to make it a little easier to read:
```
MinMaxPriorityQueue<Map.Entry<String, Integer>> pq = getEmailPriorityQueue(getEmailFreqs(itemLikes));
return FluentIterable.from(pq).transform(input -> input.getKey()).toList();
```

Reply:
I tried using the FluentIterable class from Guava but it seems that it messed up the order of the priority queue. I am not sure exactly how it is traversing the MinMaxPriorityQueue to get the order that it is giving, but I am going to stick to the method that I already have for now.